### PR TITLE
Update README with release, package, certificate information

### PR DESCRIPTION
**Pull request by olblak**
This pull request is a first attempt to document the release process.
It includes instruction for 
* releasing
* packaging
* manage code signing certificate

Feel free to highlight any missing documentation

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -1,39 +1,184 @@
...
+:toc: right
 
-As a part of the link:https://issues.jenkins-ci.org/browse/INFRA-910[INFRA-910] EPIC,
-https://jenkins.io/projects/infrastructure/[Jenkins Infrastructure Team] is working on a new Jenkins Core Release Environment
-which would be able to release Jenkins Weekly and LTS versions in an automatic way.
-It includes packaging, staging, and releasing stages.
+== Introduction
 
-## Jenkins Core Release Environment
+This repository contains everything needed to release https://github.com/jenkinsci/jenkins[Jenkins core] from Jenkins which includes:
```
Comment (MarkEWaite) [inline README.adoc:6]: Minor phrasing
```suggestion
This repository contains everything needed to release https://github.com/jenkinsci/jenkins[Jenkins core]:
```

File: README.adoc [hunk 2]
```diff
@@ -1,39 +1,184 @@
...
+* Jenkinsfiles for Releasing and Packaging.
+* Kubernetes POD templates definition for releasing and packaging on Linux and Windows environments.
+* Any configuration or utility scripts
 
-Chart for Jenkins Core Release Environment environment is available link:https://github.com/jenkins-infra/charts/blob/master/helmfile.d/jenkins-release.yaml[here].
-Jenkins master chart for it is available https://github.com/jenkins-infra/charts/tree/master/charts/jenkins[here].
+The release process implies following Git repository:
 
-## This repository
+* https://github.com/jenkinsci/jenkins[jenkinsci/jenkins] or any other git repository containing the Jenkins code that needs to be released and or packaged for.
```
Comment (MarkEWaite) [inline README.adoc:14]: Simplify phrasing
```suggestion
* https://github.com/jenkinsci/jenkins[jenkinsci/jenkins] or any other git repository containing the Jenkins code to be released.
```

File: README.adoc [hunk 3]
```diff
@@ -1,39 +1,184 @@
...
+* https://github.com/jenkinsci/packaging[jenkinsci/packaging] contains every tools and configuration used to package Jenkins for OS distribution [Debian,Redhat,Suse,MSI,WAR]
```
Comment (MarkEWaite) [inline README.adoc:15]: Correct names of operating system distributions
```suggestion
* https://github.com/jenkinsci/packaging[jenkinsci/packaging] contains tools and configuration to package Jenkins for OS distribution [Debian, Red Hat, SUSE, MSI, WAR]
```

File: README.adoc [hunk 4]
```diff
@@ -1,39 +1,184 @@
...
+* https://github.com/jenkins-infra/chart[jenkins-infra/chart] contains Helm chart used to deploy release.ci.jenkins.io on top of Kubernetes.
+* https://github.com/jenkins-infra/azure[jenkins-infra/azure] contains the terraform code used to provision infrastructures needed by the release environment
```
Comment (MarkEWaite) [inline README.adoc:17]:
```suggestion
* https://github.com/jenkins-infra/azure[jenkins-infra/azure] contains the terraform code used to provision infrastructure needed by the release environment
```

File: README.adoc [hunk 5]
```diff
@@ -1,39 +1,184 @@
...
+|`MAVEN_REPOSITORY_NAME`
+| Defines the maven repository name where it pushes generated artifacts
+
+|===
+
+=== Steps
+
+Estimated time +- 1h30
+
+. Open your favorite browser to https://release.ci.jenkins.io
```
Comment (MarkEWaite) [inline README.adoc:58]:
```suggestion
. Connect to the Jenkins VPN
. Open your favorite browser to https://release.ci.jenkins.io
```

File: README.adoc [hunk 6]
```diff
@@ -1,39 +1,184 @@
...
+. Trigger the release job on the master branch. https://release.ci.jenkins.io/blue/organizations/jenkins/core-release/branches[Link]
+. Once triggered, it asks you which release line you want to do. It's important to know that the release line matches one of the profile defines https://github.com/jenkins-infra/release/tree/master/profile.d[here], so please carefully review the settings and be sure that is what you are looking for.
```
Comment (MarkEWaite) [inline README.adoc:60]:
```suggestion
. Once triggered, it asks you which release line you want to do. It's important to know that the release line matches one of the profiles defined https://github.com/jenkins-infra/release/tree/master/profile.d[here], so please carefully review the settings and be sure that is what you are looking for.
```

File: README.adoc [hunk 7]
```diff
@@ -1,39 +1,184 @@
...
+Currently, we do not provide an automated way to promote a release but you can still stage a release by specifying a staging git repository and/or a different maven repository. Then you have to promote manually your release.
+
+To promote your git repository, you have to open a pull request from your git repository to https://github.com/jenkinsci/jenkins/compare?expand=1[jenkinsci/jenkins] on the appropriated branch.
+
+To promote your maven artifacts, you have to manually move them to the right location.
+
+
+== Package
+
+The packaging process looks after the latest Jenkins version published on a Maven repository and then build and publish artifacts for Debian, Redhat, Suse, Windows and also republish the War file on the package server.
```
Comment (MarkEWaite) [inline README.adoc:76]:
```suggestion
The packaging process looks after the latest Jenkins version published on a Maven repository and then build and publish artifacts for Debian, Red Hat, SUSE, Windows and also republishes the War file on the package server.
```

File: README.adoc [hunk 8]
```diff
@@ -1,39 +1,184 @@
...
+| RELEASELINE
+| Define the release line used by packaging scripts in https://github.com/jenkinsci/packaging[jenkinsci/packaging]
+
+|===
+
+=== Steps
+
+Estimated time +- 30min
+
+. Open your favorite browser to https://release.ci.jenkins.io[release.ci.jenkins.io]
```
Comment (MarkEWaite) [inline README.adoc:122]:
```suggestion
. Connect to the Jenkins VPN
. Open your favorite browser to https://release.ci.jenkins.io[release.ci.jenkins.io]
```

File: README.adoc [hunk 9]
```diff
@@ -1,39 +1,184 @@
...
+Once the job is done, every package will be published and then mirror synced
+
+=== Validate
+Ensure that packages are correctly published on pkg.jenkins.io and correctly signed.
+
+=== Stage
+Staging packages is not supported.
+
+== Certificate
+The Jenkins project uses a Digicert account provided by CDF to request code signing certificate.
```
Comment (MarkEWaite) [inline README.adoc:135]:
```suggestion
The Jenkins project uses a Digicert account provided by the Continuous Delivery Foundation to request code signing certificates.
```

**Comment by timja** [comment]
Is it worth mentioning anywhere how the 3 main release flows work?

Weekly - automated cron
LTS - release officer
Security - ?

**Comment by olblak** [comment]
> Looks great, worth mentioning https://github.com/jenkins-infra/docker-packaging somewhere?

What do you think to add a link in this README but explain more in detail on [jenkinsci/packaging](https://github.com/jenkinsci/packaging)

**Comment by olblak** [comment]
> Weekly - automated cron
> LTS - release officer
> Security - release or security / both? or does this go somewhere else / not supported yet?

It's not yet clear to me, who should do those releases and I would like to have a broader discussion with the people involved like @daniel-beck  or @olivergondza , I'll organise a call to discuss about it

**Comment by timja** [comment]
> > Looks great, worth mentioning https://github.com/jenkins-infra/docker-packaging somewhere?
> 
> What do you think to add a link in this README but explain more in detail on [jenkinsci/packaging](https://github.com/jenkinsci/packaging)

sounds good

**Comment by olblak** [comment]
Small glitch my two previous commits, I prefer to open a separated pull request for them

**Pull request merged**